source/simulator/assembly.cc

```diff
@@ -1497,6 +1496,31 @@ namespace aspect
           }
       }
 
+#ifdef DEBUG
+    // make sure that if the model does not use operator splitting,
+    // the material model outputs do not fill the reaction_rates (because the reaction_terms are used instead)
+    if (!parameters.use_operator_splitting)
+      {
+        const unsigned int n_q_points = scratch.finite_element_values.n_quadrature_points;
+        material_model->create_additional_named_outputs(scratch.material_model_outputs);
+        MaterialModel::ReactionRateOutputs<dim> *reaction_rate_outputs
+          = scratch.material_model_outputs.template get_additional_output<MaterialModel::ReactionRateOutputs<dim> >();
+
+        if (reaction_rate_outputs != NULL)
```

Comment:
Is this check still useful with the recent changes? If we enforce that material models only create these outputs if we use operator splitting, we would just need to ``assert( reaction_rate_outputs==NULL)`` here. Right? Sorry for the back and forth.